src/DynamicData.Tests/Utilities/TestSourceCache.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+
+using DynamicData.Kernel;
+
+namespace DynamicData.Tests.Utilities;
+
+public sealed class TestSourceCache<TObject, TKey>
+        : ISourceCache<TObject, TKey>
+    where TObject : notnull
+    where TKey : notnull
+{
+    private readonly IObservable<int> _countChanged;
```

Comment:
Could you just expose a single `Subject<IChangeSet<TObject, TKey>>` and then Connect could merge that subject with the actual Connect?

Then you could even add a method to inject ChangeSets into the stream.

I don't think you need two subjects because each Subject already handles OnComplete and OnError, I think.